File: glasses-management-backend-h2/src/main/java/com/glasses/entity/SalesRecord.java
package com.glasses.entity;

import com.mybatisflex.annotation.Column;
import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.math.BigDecimal;
import java.util.Date;

@Data
@Table("sales_record")
public class SalesRecord {
    @Id(keyType = KeyType.Auto)
    private Long id;

    private String recordNo;
    private Long customerId;
    private Long optometryId;

    @Column(ignore = true)
    private String customerName;

    // 镜架信息
    private String frameBrand;
    private String frameModel;
    private Integer frameQuantity;
    private BigDecimal frameRetailPrice;
    private BigDecimal framePrice;

    // 镜片信息
    private String lensBrand;
    private String lensParams;
    private Integer lensQuantity;
    private BigDecimal lensRetailPrice;
    private BigDecimal lensPrice;

    private BigDecimal totalRetailPrice;
    private BigDecimal totalAmount;
    
    private String remark;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private Date salesDate;
    private Long operatorId;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private Date createTime;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private Date updateTime;

    @Column(isLogicDelete = true)
    private Boolean deleted = false;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private Date deletedTime;
    private Long deletedBy;
}

File: glasses-management-android-native/app/src/main/java/com/glasses/app/model/SalesRecord.kt
package com.glasses.app.data.model

import java.math.BigDecimal

data class SalesRecord(
    val id: Long? = null,
    val recordNo: String? = null,
    val customerId: Long? = null,
    val optometryId: Long? = null,
    val customerName: String? = null,
    val frameBrand: String? = null,
    val frameModel: String? = null,
    val frameQuantity: Int? = null,
    val frameRetailPrice: BigDecimal? = null,
    val framePrice: BigDecimal? = null,
    val lensBrand: String? = null,
    val lensParams: String? = null,
    val lensQuantity: Int? = null,
    val lensRetailPrice: BigDecimal? = null,
    val lensPrice: BigDecimal? = null,
    val totalRetailPrice: BigDecimal? = null,
    val totalAmount: BigDecimal? = null,
    val remark: String? = null,
    val salesDate: String? = null,
    val operatorId: Long? = null,
    val createTime: String? = null,
    val updateTime: String? = null,
    val deleted: Boolean? = null
)
